### PR TITLE
Added model test for BuyerReview, fix #23

### LINKED ISSUE #23: Add unit tests (model and controller) for BuyerReview

### PR DESCRIPTION
**Pull request by rocka0**

### CODE REVIEW

**Review by aryanarora180 — DISMISSED** [review]
Looks good, just add another test to check data validation.

File: src/controllers/buyerReview.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ export const getBuyerReviewById = async (_id: string) => {
   }
 };
 
-export const getBuyerReviewsByBuyerUser = async (buyer_id: string) => {
+export const getBuyerReviewsBymockBuyerUser = async (buyer_id: string) => {
```
Comment (aryanarora180) [inline src/controllers/buyerReview.ts]: why lol

File: src/controllers/buyerReview.ts [hunk 2]
```diff
@@ -62,11 +62,11 @@ export const getBuyerReviewsByBuyerUser = async (buyer_id: string) => {
   }
 };
 
-export const getBuyerReviewsByPostingUser = async (user_id: string) => {
+export const getBuyerReviewsBymockSellerUser = async (user_id: string) => {
```
Comment (aryanarora180) [inline src/controllers/buyerReview.ts]: again, why?

File: src/controllers/buyerReview.ts [hunk 3]
```diff
@@ -62,11 +62,11 @@ export const getBuyerReviewsByBuyerUser = async (buyer_id: string) => {
   }
 };
 
-export const getBuyerReviewsByPostingUser = async (user_id: string) => {
```
Comment (aryanarora180) [inline src/controllers/buyerReview.ts:65]: A route for BuyerReview (api/reviews/buyer/byuser/:userId) still uses the function

**Review by aryanarora180 — DISMISSED** [review]

File: tests/controllers/buyerReview.test.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+      await mongoServer.stop();
+    }
+  });
+
+  it('Should be able to fetch all BuyerReviews', async () => {
+    let getAllBuyerReviewsQuery = await getAllBuyerReviews();
+    expect(getAllBuyerReviewsQuery.status).toBe(true);
+    expect(getAllBuyerReviewsQuery.data).toBeInstanceOf(Array);
+    expect(getAllBuyerReviewsQuery.data?.length).toBe(1);
+    expect(new ObjectId(getAllBuyerReviewsQuery.data?.at(0)?._id)).toEqual(
```
Comment (aryanarora180) [inline tests/controllers/buyerReview.test.ts:52]: No need for new ObjectId()

File: tests/controllers/buyerReview.test.ts [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    );
+  });
+
+  it('Should be able to fetch BuyerReview by ID', async () => {
+    let getBuyerReviewByIdQuery = await getBuyerReviewById(
+      mockBuyerReview._id.toString()
+    );
+    expect(getBuyerReviewByIdQuery.status).toBe(true);
+    expect(getBuyerReviewByIdQuery.data).toBeInstanceOf(BuyerReview);
+    expect(new ObjectId(getBuyerReviewByIdQuery.data?._id)).toEqual(
```
Comment (aryanarora180) [inline tests/controllers/buyerReview.test.ts:63]: There's no need for new ObjectId in any of the tests. You cna directly say `expect(getBuyerReviewByIdQuery.data?._id).toEqual(mockBuyerReview._id)`

File: tests/models/buyerReview.test.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    expect(pageHitInDb?.request).toEqual(mockBuyerRequest._id);
+    expect(pageHitInDb?.buyer).toEqual(mockBuyerUser._id);
+    expect(pageHitInDb?.seller).toEqual(mockSellerUser._id);
+    expect(pageHitInDb?.service).toEqual(mockSellerService._id);
+    expect(pageHitInDb?.comment).toEqual('Nice service');
+    expect(pageHitInDb?.rating).toEqual(4.2);
+  });
+
+  it('Should not accept invalid data', async () => {
+    const invalidBuyerReview = new BuyerReview({
```
Comment (aryanarora180) [inline tests/models/buyerReview.test.ts:59]: This seems to be a valid BuyerReview

File: tests/controllers/buyerReview.test.ts [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+    expect(createBuyerReviewQuery.data?.rating).toEqual(3.6);
+  });
+
+  it('Should be able to delete a BuyerReview', async () => {
+    let deleteBuyerReviewQuery = await deleteBuyerReview(
+      mockBuyerReview._id.toString(),
+      mockSellerUser._id.toString()
+    );
+    expect(deleteBuyerReviewQuery.status).toBe(true);
+    expect(await BuyerReview.countDocuments({})).toEqual(1);
```
Comment (aryanarora180) [inline tests/controllers/buyerReview.test.ts:139]: You should check to see if the BuyerReview has been deleted. Even if the number of documents are 1, the BuyerReview need not have necessarily been deleted.

**Review by aryanarora180 — CHANGES_REQUESTED** [review]

File: tests/controllers/buyerReview.test.ts [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+  });
+
+  it('Should be able to fetch all BuyerReviews', async () => {
+    let getAllBuyerReviewsQuery = await getAllBuyerReviews();
+    expect(getAllBuyerReviewsQuery.status).toBe(true);
+    expect(getAllBuyerReviewsQuery.data).toBeInstanceOf(Array);
+    expect(getAllBuyerReviewsQuery.data?.length).toBe(1);
+    expect(getAllBuyerReviewsQuery.data?.at(0)?._id).toEqual(
+      mockBuyerReview._id
+    );
```
Comment (aryanarora180) [inline tests/controllers/buyerReview.test.ts:53]: I would recommend inserting another mock and then also checking if 2 BuyerReviews have been returned.

File: tests/models/buyerReview.test.ts (hunk 6 above)
Comment (rocka0) [reply tests/models/buyerReview.test.ts:59]: Rating is > `5`

File: tests/controllers/buyerReview.test.ts (hunk 7 above)
Comment (rocka0) [reply tests/controllers/buyerReview.test.ts]: Done

File: tests/controllers/buyerReview.test.ts (hunk 8 above)
Comment (rocka0) [reply tests/controllers/buyerReview.test.ts:54]: Done

File: tests/models/buyerReview.test.ts (hunk 6 above)
Comment (aryanarora180) [reply tests/models/buyerReview.test.ts:59]: f mb lmao

**Pull request merged**